python/cudf/cudf/core/reshape.py

```diff
@@ -278,9 +327,25 @@ def concat(objs, axis=0, ignore_index=False, sort=None):
                 result = objs[0].copy()
             return result
         else:
-            return cudf.DataFrame._concat(
-                objs, axis=axis, ignore_index=ignore_index, sort=sort
+            result = cudf.DataFrame._concat(
+                objs,
+                axis=axis,
+                join=join,
+                ignore_index=ignore_index,
+                sort=sort,
             )
+            if ignore_index and join == "inner" and result.empty:
+                # this occurs with no overlapping column names
+                result.index = cudf.RangeIndex(sum(len(obj) for obj in objs))
+                return result
+            else:
+                return cudf.DataFrame._concat(
+                    objs,
+                    axis=axis,
+                    join=join,
+                    ignore_index=ignore_index,
+                    sort=sort,
+                )
```

Comment:
You seem to be calling `cudf.DataFrame._concat` in line 330 Why do we call it a second time again?

Reply:
Whoops, thanks for the catch here. I hadn't noticed I was calling it twice. I'll resolve this :D